lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -3164,7 +3182,8 @@ ActorIsolation ActorIsolationRequest::evaluate(
     }
 
     if (auto nominal = value->getDeclContext()->getSelfNominalTypeDecl()) {
-      if (nominal->isDistributedActor()) {
+      /// Unless the function is static, it is isolated to the dist actor
+      if (nominal->isDistributedActor() && !func->isStatic()) {
```

Comment:
This is a bit nasty that we had to check ad-hoc here, i'll rethink our isolation in general...